orc8r/gateway/python/magma/magmad/proxy_client.py

```diff
@@ -95,11 +95,12 @@ async def send(self, gateway_request, req_id, sync_rpc_response_queue,
                 SyncRPCResponse(heartBeat=False, reqId=req_id,
                                 respBody=GatewayResponse(err=str(e))))
         finally:
-            del self._connection_table[req_id]
             client.close_connection()
+            del self._connection_table[req_id]
 
     def close_all_connections(self):
-        for _, client in self._connection_table.items():
+        connections = list(self._connection_table.values())
+        for client in connections:
             client.close_connection()
```

Comment:
Should we wrap this around try clause, What happens when the connection list has a stale connection value?

Reply:
This shouldn't be a problem as the connections to be closed should be all valid (part of the connection_table), I'll add the check anyway just in case as should help with debugging further cases.